Gems/Atom/Tools/AtomToolsFramework/Code/Source/DynamicNode/DynamicNodeSlotConfig.cpp

```diff
@@ -97,75 +95,124 @@ namespace AtomToolsFramework
+            return dataType->GetDisplayName() == m_defaultDataType;
+        });
+
+        // If the default data type is not set or does not match any of the supported types then assign it to the same type as the default
+        // value or the first registered data type.
+        if (defaultTypeItr == supportedDataTypes.end())
+        {
+            if (defaultValueItr != supportedDataTypes.end())
             {
-                for (const auto& selection : selections)
-                {
-                    if (dataType->GetDisplayName() == selection)
-                    {
-                        m_defaultValue = dataType->GetDefaultValue();
-                        return AZ::Edit::PropertyRefreshLevels::EntireTree;
-                    }
-                }
+                m_defaultDataType = (*defaultValueItr)->GetDisplayName();
+                defaultTypeItr = defaultValueItr;
             }
+            else
+            {
+                m_defaultDataType = supportedDataTypes.front()->GetDisplayName();
+                defaultTypeItr = supportedDataTypes.begin();
+            }
+        }
 
-            return AZ::Edit::PropertyRefreshLevels::EntireTree;
+        // Finally, if the default value does not match the default data type then reset it.
+        if (defaultTypeItr != defaultValueItr)
+        {
+            m_defaultValue = (*defaultTypeItr)->GetDefaultValue();
         }
 
-        return AZ::Edit::PropertyRefreshLevels::AttributesAndValues;
+        return AZ::Edit::PropertyRefreshLevels::EntireTree;
     }
 
-    AZ::Crc32 DynamicNodeSlotConfig::ClearDefaultValue()
+    AZStd::any DynamicNodeSlotConfig::GetDefaultValue() const
     {
-        m_defaultValue.clear();
-        return AZ::Edit::PropertyRefreshLevels::EntireTree;
+        const auto& supportedDataTypes = GetSupportedDataTypes();
+        for (const auto& dataType : supportedDataTypes)
+        {
+            if (dataType->IsSupportedValue(m_defaultValue))
+            {
+                return m_defaultValue;
+            }
+        }
+        return !supportedDataTypes.empty() ? supportedDataTypes.front()->GetDefaultValue() : AZStd::any();
     }
 
-    AZ::Crc32 DynamicNodeSlotConfig::ClearDefaultValueIfInvalid()
+    AZStd::string DynamicNodeSlotConfig::GetDefaultDataTypeName() const
     {
-        GraphModel::DataTypeList registeredDataTypes;
-        DynamicNodeManagerRequestBus::BroadcastResult(registeredDataTypes, &DynamicNodeManagerRequestBus::Events::GetRegisteredDataTypes);
+        const auto& supportedDataTypes = GetSupportedDataTypes();
+        for (const auto& dataType : supportedDataTypes)
+        {
+            if (dataType->GetDisplayName() == m_defaultDataType)
+            {
+                return m_defaultDataType;
+            }
+        }
+        return !supportedDataTypes.empty() ? supportedDataTypes.front()->GetDisplayName() : AZStd::string();
```

Comment:
Can this just call GetDefaultDataType()->GetDisplayName()?

Reply:
Yes! It's always simplified compared to what I had before but that's yet another simplification.